packages/rtk-query-codegen-openapi/test/generateEndpoints.test.ts

```diff
@@ -0,0 +1,179 @@
+  expect(api).toContain('useAddPetMutation');
+  expect(api).toMatchSnapshot(
+    'should generate an `useGetPetByIdQuery` query hook and an `useAddPetMutation` mutation hook'
+  );
+});
+
+test('hooks generation uses overrides', async () => {
+  const api = await generateEndpoints({
+    apiFile: './fixtures/emptyApi.ts',
+    schemaFile: resolve(__dirname, 'fixtures/petstore.json'),
+    filterEndpoints: 'loginUser',
+    endpointOverrides: [
+      {
+        pattern: 'loginUser',
+        type: 'mutation',
+      },
+    ],
+    hooks: true,
+  });
+  expect(api).not.toContain('useLoginUserQuery');
+  expect(api).toContain('useLoginUserMutation');
+  expect(api).toMatchSnapshot('should generate an `useLoginMutation` mutation hook');
+});
+
+test('should use brackets in a querystring urls arg, when the arg contains full stops', async () => {
+  const api = await generateEndpoints({
+    apiFile: './fixtures/emptyApi.ts',
+    schemaFile: resolve(__dirname, 'fixtures/params.json'),
+  });
+  // eslint-disable-next-line no-template-curly-in-string
+  expect(api).toContain('`/api/v1/list/${queryArg["item.id"]}`');
+  expect(api).toMatchSnapshot();
+});
+
+test('apiImport builds correct `import` statement', async () => {
+  const api = await generateEndpoints({
+    apiFile: './fixtures/emptyApi.ts',
+    schemaFile: resolve(__dirname, 'fixtures/params.json'),
+    filterEndpoints: [],
+    apiImport: 'myApi',
+  });
+  expect(api).toContain('myApi as api');
+});
+
+describe('import paths', () => {
+  test('should create paths relative to `outFile` when `apiFile` is relative (different folder)', async () => {
+    process.chdir(__dirname);
+    await generateEndpoints({
+      apiFile: './fixtures/emptyApi.ts',
+      outputFile: './tmp/out.ts',
+      schemaFile: resolve(__dirname, 'fixtures/petstore.json'),
+      filterEndpoints: [],
+      hooks: true,
+    });
+    expect(await fs.promises.readFile('./tmp/out.ts', 'utf8')).toContain("import { api } from '../fixtures/emptyApi'");
+  });
+
+  test('should create paths relative to `outFile` when `apiFile` is relative (same folder)', async () => {
+    process.chdir(__dirname);
+
+    await fs.promises.writeFile('./tmp/emptyApi.ts', await fs.promises.readFile('./fixtures/emptyApi.ts'));
+
+    await generateEndpoints({
+      apiFile: './tmp/emptyApi.ts',
+      outputFile: './tmp/out.ts',
+      schemaFile: resolve(__dirname, 'fixtures/petstore.json'),
+      filterEndpoints: [],
+      hooks: true,
+    });
+    expect(await fs.promises.readFile('./tmp/out.ts', 'utf8')).toContain("import { api } from './emptyApi'");
+  });
+});
+
+describe('yaml parsing', () => {
+  it.skip('should parse a yaml schema from a URL', async () => {
+    // TODO
+    const result = await generateEndpoints({
+      apiFile: './tmp/emptyApi.ts',
+      schemaFile: `https://petstore3.swagger.io/api/v3/openapi.yaml`,
+      hooks: true,
+    });
+    expect(result).toMatchSnapshot();
+  });
```

Comment:
@msutkowski could you look into this one? That really seems to be broken, but I have no idea why - you added that back then. 

It works with `.json` o_O

Reply:
ping @msutkowski